src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_validators.py

```diff
@@ -193,11 +261,26 @@ def _validate_vm_create_vnet(namespace):
             # 2 - user specified existing vnet/subnet
             namespace.vnet_type = 'existing'
             return
-
     # 3 - create a new vnet/subnet
     namespace.vnet_type = 'new'
 
 
+def _validate_vmss_create_subnet(namespace):
+    # TODO: we can consider for non-new sceanrio there are user asks
```

Comment:
no, this is a new thing that we could consider, that we should check whether the subnet is big enough to contain the the number of instances. if not, we should create a new one rather reuse it